src/segment-loader.js

```diff
@@ -2031,6 +2034,27 @@ export default class SegmentLoader extends videojs.EventTarget {
     // as we use the start of the segment to offset the best guess (playlist provided)
     // timestamp offset.
     this.updateSourceBufferTimestampOffset_(segmentInfo);
+
+    // throw away the isSyncRequest segment if
+    // it wouldn't have been the next segment we request.
+    if (segmentInfo.isSyncRequest) {
+      this.syncController_.saveSegmentTimingInfo({
```

Comment:
Should we call this function within this block? From a brief glance it seems like it only estimates the segment end point using start/duration unless we have an `end`. Seems like we should call it here.

Reply:
I think we should probably call it. On that note we may also want to consider adding a property to the segment whether the last time we requested that segment it was a sync request (and clear it if it's requested on a non sync request). It may help us for debugging, particularly now that it's no longer being appended.

Reply:
called